clj-refactor.el

```diff
@@ -2181,18 +2193,34 @@ possible choices. If the choice is trivial, return it."
         (completing-read prompt choices nil nil nil nil (car choices)))
     (read-from-minibuffer prompt)))
 
+(defun cljr--insert-into-project-dependencies (artifact version)
+  (re-search-forward ":dependencies")
+  (paredit-forward)
+  (paredit-backward-down)
+  (newline-and-indent)
+  (insert "[" artifact " \"" version "\"]"))
+
+(defun cljr--insert-into-project-deps (artifact version)
+  (re-search-forward ":deps")
+  (forward-sexp)
+  (backward-char)
+  (newline-and-indent)
+  (insert artifact " {:mvn/version \"" version "\"}"))
+
 (defun cljr--add-project-dependency (artifact version)
-  (cljr--update-file (cljr--project-file)
-    (goto-char (point-min))
-    (re-search-forward ":dependencies")
-    (paredit-forward)
-    (paredit-backward-down)
-    (newline-and-indent)
-    (insert "[" artifact " \"" version "\"]")
-    (cljr--post-command-message "Added %s version %s as a project dependency" artifact version)
-    (when cljr-hotload-dependencies
-      (paredit-backward-down)
-      (cljr-hotload-dependency))))
+  (let* ((project-file (cljr--project-file))
+         (depsp (cljr--project-with-deps-p project-file)))
+    (cljr--update-file project-file
+      (goto-char (point-min))
+      (if depsp
+          (cljr--insert-into-project-deps artifact version)
+        (cljr--insert-into-project-dependencies artifact version))
+      (cljr--post-command-message "Added %s version %s as a project dependency" artifact version)
+      (when cljr-hotload-dependencies
+        (if depsp
+            (backward-sexp 2)
+          (paredit-backward-down))
+        (cljr-hotload-dependency)))))
```

Comment:
This seems unlikely to work out of the box given the different data format.

Am I wrong?